### DISCUSSION TITLE
Multiple drone cooperative transport - require advice

### DISCUSSION THREAD

**Discussion by Vrinda73** [bitcraze/discussions#1176] [Q&A]
Hello!
I want to implement cooperative transport using two drones on a stick with rigid connections. I do not have access to a mocap system in our organization. Thus, I was planning to use flow deck beneath the Crazyflies and have a leader slave algorithm with velocity commands. I wanted to receive some guidance regarding this. Hopefully I am on the right track. Please note that I have tried with attitude commands but the system is very difficult to tune. Please see the picture below for your reference.
![image](https://github.com/bitcraze/discussions/assets/44201361/3b9a01ce-97f1-4771-bfd2-795d8efbbd19)

**Comment by knmcguire** [comment]
Hi! 

Sure you can use the flowdeck for that, as long as you are aware that both drones are not sharing the same coordinate system (where-ever they start that is where their zero is). The flow deck is a relative positioning system and they also will drift apart in time, but perhaps that will be okay if you'll just control it with velocity commands. But I'm not sure if they will fly like that out of the box, as their dynamics are now linked to eachother. There have been some papers written on this control problem alone.

Here some links that might be handy:
- Flowdeck tips&tricks: https://www.bitcraze.io/2023/11/go-with-the-flow-relative-positioning-with-the-flow-deck/
- Positioning systems (maybe lighthouse might be a good option for you?) : https://www.bitcraze.io/2021/05/positioning-system-overview/
- Research page with publication: https://www.bitcraze.io/portals/research/#publications

**Reply by Vrinda73** [reply]
I have been trying to fly this system cooperatively without much success. Little demotivated due to it. I hope someone gives me key insights in order to solve it. So currently all I am trying to do is to make the system hover using central controller. So both the drones have same thrust command. I have tried HL commander to give position waypoints and velocity commands but the system crashes. I can only see the system flying with attitude and thrust commands. But, getting the thrust to hold the height is so challenging. I am implementing a PD controller which provides desired accelration in x, y and z viz. ax_des, ay_des and az_des. My first controller for thrust control for each drone was,
`thrust_cmd = thrust_hover + int(k*az_des)`
Values thrust_hover = 44000 and k = 10000. Note that thrust_cmd is sent to both the drones. In this controller, the system is seen to oscillate near the ground and never takes off even after increasing thrust_hover or az_des. I then tried to give higher thrust_cmd only initially such as 47000 for few iterations, to overcome ground effect, but the system flies high and again hovers near ground. Recently, I came across mellinger controller and saw that the thrust should account for commanded roll and pitch commands. Thus I tried, 
`thrust_cmd = massthrust*np.dot(mass_CF*(ades + g*e3), R.e3)`
where e3 = [ 0 0 1] and massthrust = 132000, but eversince the system has been crashing. I tried changing massthrust value with no luck. I have referred to recent publications [https://www.bitcraze.io/2023/11/adaptive-cooperative-flight-for-transportation-using-crazyflies/](url) and [https://ieeexplore.ieee.org/abstract/document/8461014?casa_token=01dlMMwl0E4AAAAA:XUCwNO3ow6kKr8UvN4dF1I7hPasyKuvoH-Mm0GP9vZc1QIkGxsEpWWWvPzAgbPG_IqyBiizC](url) however, how exactly thrust is controlled is unclear. 

I am aiming for ICRA deadline and any help is appreciated. 

Thank you!
